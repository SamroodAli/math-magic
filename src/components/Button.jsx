import React from 'react';

class Button extends React.PureComponent {
  render() {
    return (
      <h1>Hello world</h1>
    );
  }
}

export default Button;
