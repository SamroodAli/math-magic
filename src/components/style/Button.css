.button {
  width: 25%;
  display: flex;
  justify-content: center;
  align-items: center;
}
