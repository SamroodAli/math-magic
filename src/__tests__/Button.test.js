import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Button from '../components/Button';

test('Button component returns a button react element', async () => {
  const onClick = jest.fn();
  const buttonRender = render(<Button name="1" onClick={onClick} />);
  const button = await buttonRender.findByTestId('button');
  expect(button.type).toBe('button');
});

test('Button component to call the number with the onClick function', async () => {
  const onClick = jest.fn();
  const buttonRender = render(<Button name="1" onClick={onClick} />);
  const button = await buttonRender.findByTestId('button');
  button.click();
  expect(onClick).toHaveBeenCalledWith('1');
});

test('Button component should be wide if wide prop given', async () => {
  const onClick = jest.fn();
  const buttonRender = render(<Button name="1" onClick={onClick} wide />);
  const button = await buttonRender.findByTestId('button');
  expect(button.style.width).toEqual('50%');
});

test('Button component should not be wide if wide prop not given', async () => {
  const onClick = jest.fn();
  const buttonRender = render(<Button name="1" onClick={onClick} />);
  const button = await buttonRender.findByTestId('button');
  expect(button.style.width).toEqual('25%');
});

test('Button component should be orange if color prop not given', async () => {
  const onClick = jest.fn();
  const buttonRender = render(<Button name="1" onClick={onClick} />);
  const button = await buttonRender.findByTestId('button');
  expect(button.style.backgroundColor).toEqual('orange');
});

test('Button component should be given color if color prop given', async () => {
  const onClick = jest.fn();
  const buttonRender = render(<Button name="1" onClick={onClick} color="grey" />);
  const button = await buttonRender.findByTestId('button');
  expect(button.style.backgroundColor).toEqual('grey');
});

test('Button snapshot testing', () => {
  const tree = renderer
    .create(<Button name="1" onClick={jest.fn()} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
