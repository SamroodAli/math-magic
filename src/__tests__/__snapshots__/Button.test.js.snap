// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot testing 1`] = `
<button
  className="button"
  data-testid="button"
  onClick={[Function]}
  type="button"
>
  1
</button>
`;
